app/src/main/java/com/pavesid/androidacademy/ui/MainActivity.kt

```diff
@@ -1,19 +1,62 @@
 package com.pavesid.androidacademy.ui
 
+import android.hardware.Sensor
+import android.hardware.Sensor.TYPE_ACCELEROMETER
+import android.hardware.Sensor.TYPE_MAGNETIC_FIELD
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager
+import android.os.Build
 import android.os.Bundle
+import android.view.Display
 import android.view.MenuItem
+import android.view.Surface
+import android.view.WindowManager
 import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
 import com.pavesid.androidacademy.R
 import com.pavesid.androidacademy.databinding.ActivityMainBinding
 import com.pavesid.androidacademy.ui.details.MoviesDetailsFragment
 import com.pavesid.androidacademy.ui.movies.MoviesFragment
+import kotlin.math.atan2
+import kotlin.math.round
+import kotlin.math.sqrt
 
-class MainActivity : AppCompatActivity(), MoviesFragment.Listener {
+class MainActivity : AppCompatActivity(), MoviesFragment.Listener, SensorEventListener {
 
     private var _binding: ActivityMainBinding? = null
     private val binding: ActivityMainBinding
```

Comment:
could you please tell me what the reason to use 2 fields for binding?

Reply:
https://developer.android.com/topic/libraries/view-binding#fragments

Reply:
Done.